app/models/vim_performance_state/purging.rb

```diff
@@ -8,11 +8,20 @@ def purge_mode_and_value
         %w[orphaned resource]
       end
 
-      # remove anything where the resource no longer exists AND
-      # remove anything older than a certain date
+      # remove anything older than a certain date AND
+      # remove anything where the resource no longer exists
+      # Use a callback to ensure :date is run first, before
+      # :orphaned and not concurrently.
       def purge_timer
+        callback = {
+          :class_name  => name,
+          :method_name => :purge_callback
+        }
+        purge_queue(:date, purge_date, callback)
```

Comment:
Can you pass args to the callback? If so, this might be cleaner to just call it directly instead of through a new method...something like

```suggestion
        callback = {
          :class_name  => name,
          :method_name => :purge_queue
          :args        => [:orphaned, "resource"]
        }
        purge_queue(:date, purge_date, callback)
```

Reply:
Also the "after" being visually first is oddly confusing - wondering if this is more readable:

```suggestion
        purge_queue(:date, purge_date, miq_callback: {
          :class_name  => name,
          :method_name => :purge_queue
          :args        => [:orphaned, "resource"]
        })
```

and then change the `purge_queue` signature to have a keyword arg for the callback.

Reply:
No, that's what I tried at first.  You can see it in the logging, it's meant to pass along the status of the prior work item to the callback:

```
[----] I, [2025-03-20T14:10:39.888159#16662:88cc]  INFO -- evm: MIQ(MiqQueue#m_callback) Message id: [25981200], Invoking Callback with args: ["ok", "Message delivered successfully", "844406"]
```

Reply:
I can inline the callback.  Let me see how that looks.

Reply:
Ok, pushed.  Not sure if this is more straightforward than https://github.com/ManageIQ/manageiq/pull/23385